### DISCUSSION TITLE
Smart-doc RoadMap

### DISCUSSION THREAD

**Discussion by shalousun** [TongchengOpenSource/smart-doc#640] [Announcements]
## User Cases
Welcome to contribute user cases to the community.
欢迎给社区贡献用户案例

## Feature
- [x] gRPC Protocol Documentation Support #851 
- [x] WebSocket Protocol Support #752 
- [x] Word Format Document Generation #686 
- [x] JMeter Performance Test Script Generation #668 
- [x] Incremental Interface Push Based on Git Management #617 
- [x] Support for New Features in Higher Versions of JDK
- [ ] Support for Reading External Markdown Templates
- [x] Support for Custom Extension Parsing, External Implementation of smart-doc Plugins #839
- [ ] Idea Plugin
- [ ] Native Installation of Smart-Doc, Direct Command Line Invocation
- [ ] Support for other languages, such as smart-doc-go

**Comment by shalousun** [comment]
Students who want to contribute can comment below this discussion. If you confirm to claim a function, you can create an issue first
想要贡献的同学可以在本讨论下评论，如果确认认领某个功能后可以先新建一个issue

**Comment by chenqi146** [comment]
Regarding point six (reading external markdown templates), there seems to be no documentation explaining the contents of our template parameters. Is it necessary to replace parameters in external templates?
对于第六点(读取外部markdown模板), 目前好像没有文档说明我们模板参数内容, 外部模板是否需要进行参数替换呢

**Reply by chenqi146** [reply]
1. Fill at the beginning of the entire page
2. Fill at the beginning of each module
